cuda_core/cuda/core/experimental/_program.py

```diff
@@ -6,6 +6,157 @@
 from cuda.core.experimental._utils import handle_return
 from cuda.core.experimental._module import ObjectCode
 
+from typing import Optional, Tuple, Union
+from dataclasses import dataclass
+    
+@dataclass
+class ProgramOptions:
```

Comment:
TODO: 
1. As we discussed briefly offline, let's find out if we can autogen the docstrings. It should contain the description for the option, the default value if unset, acceptable value type, and then explain what the underlying C option it maps to, something like (no need to take this literally)
```
gpu_architecture : Specify the name of the class of GPU architectures for which the input must be compiled.

.. seealso:: ` --gpu-architecture`, `-arch`
```
2. There are some options we might have to tweak/combine, e.g. `device_c ` vs `relocatable_device_code`. Let's chat offline about this.
3. For `gpu_architecture` I think we should detect the current device's CC if it's unset, instead of falling back to NVRTC default

Reply:
@leofang, Are there any other options you think I should revisit? I'm hesitant to have arch fall back to the current device since there is no requirement to have called set_device by compile time.

Reply:
> since there is no requirement to have called set_device by compile time

Technically, no, but practically it would be highly unusual if one does not already have a device set to current when JIT compiling stuff. 

That said, have you noticed that calling `Device.compute_capability` does not require any CUDA context? 🙂 
```python
>>> from cuda.core.experimental import Device
>>> dev = Device()
>>> dev.compute_capability  # no need to call dev.set_current()
ComputeCapability(major=8, minor=9)
```